### ISSUE TITLE
[Bug]: import { ConsoleLoggingListener } from "./src/common.browser/Exports";

### ISSUE THREAD

**Issue by bennaaym** [microsoft/cognitive-services-speech-sdk-js#752]
### What happened?

Hello, I'm getting the following error when using the package with nextjs:

```
import { ConsoleLoggingListener } from "./src/common.browser/Exports";
SyntaxError: Cannot use import statement outside a module
```

- Environement:
```
"node": 18.15.0,
"next": "13.2.4",
"microsoft-cognitiveservices-speech-sdk": "^1.32.0",
````

- tsconfig
``` "compilerOptions": {
    "target": "ES2020",
    "module": "nodenext",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": false,
    "noEmit": true,
    "jsx": "preserve",
    "esModuleInterop": true,
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "incremental": true,
  },```


### Version

1.31.0 (Edge)

### What browser/platform are you seeing the problem on?

Chrome

### Relevant log output

_No response_

**Comment by bennaaym** [comment]
I fixed the issue by downgrading to version 1.30.0

**Comment by orgads** [comment]
This is not a fix. Please reopen.

**Comment by orgads** [comment]
1.32 is also ok. The problem appeared in 1.33.

**Comment by bennaaym** [comment]
I faced the issue with the 1.32 while using node 18.15.0 and node 18.18.0

**Comment by orgads** [comment]
Maybe you installed `^1.32`, which effectively installs `1.33`.

**Comment by bennaaym** [comment]
You are right, that was the case

**Comment by glharper** [comment]
@bennaaym Thanks for using SpeechSDK, and writing this issue up. ConsoleLoggingListener is an internal class not intended for customer use. What functionality of that class are you using that the exposed Diagnostics class does not provide?

**Comment by bennaaym** [comment]
@glharper, I m not using ConsoleLoggingListner on my code in any place these are the only functions/classes I m using in my code 
````
import * as sdk from 'microsoft-cognitiveservices-speech-sdk';
````
````
sdk.SpeechConfig.fromSubscription
sdk.AudioConfig.fromAudioFileOutput
sdk.SpeechSynthesizer
````

Also if the class is internal why is it imported to the `distrib\lib\microsoft.cognitiveservices.speech.sdk.js`

![image](https://github.com/microsoft/cognitive-services-speech-sdk-js/assets/68559207/882d37ba-5f32-47ec-8e99-e84e0bf2e23c)

**Comment by orgads** [comment]
> @bennaaym Thanks for using JS Speech SDK, and writing this issue up. ConsoleLoggingListener is an internal class not intended for customer use. What functionality of that class are you using that the exposed Diagnostics class does not provide?

Thanks @glharper. We did use `ConsoleLoggingListener`. I wasn't aware of `Diagnostics`.

Replaced it now with `Diagnostics`, and it builds fine here.

**Comment by orgads** [comment]
I take my words back :)

It builds ok with Node 20.9, but fails with Node 18.16.

**Comment by dcts** [comment]
Having the same issue. Downgrading to 1.3.0 worked for me too. Thanks @bennaaym .

**Comment by asifmai** [comment]
> Having the same issue. Downgrading to 1.3.0 worked for me too. Thanks @bennaaym .

1.3.0 or 1.30.0?

**Comment by nilsandresen** [comment]
> > Having the same issue. Downgrading to 1.3.0 worked for me too. Thanks @bennaaym .
> 
> 1.3.0 or 1.30.0?

You can use "1.32.0" without ^ and you should be fine. 1.33 is not even tagged on github, so it's probably too early to use it.

**Comment by nilsandresen** [comment]
> > Having the same issue. Downgrading to 1.3.0 worked for me too. Thanks @bennaaym .
> 
> 1.3.0 or 1.30.0?

You can use "1.32.0" without ^ and you should be fine. 1.33 is not even tagged on github, so it's probably too early to use it.

**Comment by asifmai** [comment]
> > > Having the same issue. Downgrading to 1.3.0 worked for me too. Thanks @bennaaym .
> > 
> > 
> > 1.3.0 or 1.30.0?
> 
> @asifmai You can use "1.32.0" without ^ and you should be fine. 1.33 is not even tagged on github, so it's probably too early to use it.

@nilsandresen I tried with 1.32.0 without ^, but still getting the same error. I'm using nodejs 18.16.0

**Comment by nilsandresen** [comment]
@asifmai have you checked which version is installed? Otherwise maybe try deleting node_modules and package-lock.json

**Comment by glharper** [comment]
@orgads @nilsandresen @asifmai @bennaaym @dcts I'm tracking an issue where the latest version throws in node. There is a provisional fix here: https://github.com/microsoft/cognitive-services-speech-sdk-js/pull/755, which will probably become v1.33.1 if all CI testing passes.

**Comment by TechyChan** [comment]
I'm hitting this issue as well; downgrading to 1.30.0 does not work for my use case, as I'm using `ConversationTranscriber.startTranscribingAsync` and it's throwing the error `Trace: TypeError: Cannot read properties of undefined (reading 'startContinuousRecognitionAsync')
    at ConversationTranscriber.startTranscribingAsync`

**Comment by Wiiki0807** [comment]
i used turnaround to downgrade npm verson to 1.30, then resolve.
![image](https://github.com/microsoft/cognitive-services-speech-sdk-js/assets/27147987/dd547911-a16b-4890-b932-d3c1468e4af0)

**Comment by Megamannen** [comment]
This fails with vanillajs, no tooling

minimal file
```js
const {
    SpeechConfig
} = require("microsoft-cognitiveservices-speech-sdk");

const speechConfig = SpeechConfig.fromHost(new URL('wss://spik.spik.com/'));

console.log(speechConfig);
```

```json
{
  "name": "fail-fail",
  "dependencies": {
    "microsoft-cognitiveservices-speech-sdk": "1.33"

  }
}
```

```bash
node fail.js
```

**Comment by glharper** [comment]
> This fails with vanillajs, no tooling

Understood, we should have a v1.33.1 out today or tomorrow fixing this issue.

**Comment by glharper** [comment]
@orgads @nilsandresen @asifmai @bennaaym @dcts @Megamannen @TechyChan @Wiiki0807 
v1.33.1, which is in ESM form, is available now. Note the updated samples [here](https://github.com/Azure-Samples/cognitive-services-speech-sdk/tree/master/samples/js/node) which run ESM-compliant code, demonstrating the various Speech SDK scenarios. Release notes will follow within the next few days.

**Comment by orgads** [comment]
@glharper It compiles now, but doesn't run on CJS. Is this intentional?

```
Error [ERR_REQUIRE_ESM]: require() of ES Module F:\Projects\myproject\node_modules\microsoft-cognitiveservices-speech-sdk\distrib\lib\microsoft.cognitiveservices.speech.sdk.js from F:\Projects\myproject\dist\azure-stt.js not supported.
Instead change the require of microsoft.cognitiveservices.speech.sdk.js in F:\Projects\myproject\dist\azure-stt.js to a dynamic import() which is available in all CommonJS modules.
```

**Comment by glharper** [comment]
@orgads Yes.

**Comment by orgads** [comment]
Then in my opinion this should be a major version. Minor updates are not supposed to break running systems.

**Comment by glharper** [comment]
> Then in my opinion this should be a major version. Minor updates are not supposed to break running systems.

Absolutely a fair point. The security fix we made in 1.33.0 had implications I was unaware of, and I would have added a more urgent message to the 1.32.0 release notes as a warning had I understood those implications better.

**Comment by dcts** [comment]
> > Having the same issue. Downgrading to 1.3.0 worked for me too. Thanks @bennaaym .
> 
> 1.3.0 or 1.30.0?

1.30.0, sorry was a typo. Updated my original comment.

**Comment by RodriSanchez1** [comment]
Hi! I'm using Jest to run my test. With the version 1.33.1 im having the following error: 

 Details:

    C:\Users\myproject\node_modules\microsoft-cognitiveservices-speech-sdk\distrib\lib\src\common.browser\PCMRecorder.js:85
          this.privSpeechProcessorScript = new URL( /* webpackChunkName: 'script_processor_audioWorklet' */"speech-processor.js", import.meta.url).toString();
                                                                                                                                         ^^^^

    SyntaxError: Cannot use 'import.meta' outside a module

    > 1 | import * as azureSdk from 'microsoft-cognitiveservices-speech-sdk';
        | ^

I wanted to share this with you so you are aware! @glharper

**Comment by glharper** [comment]
> Hi! I'm using Jest to run my test. With the version 1.33.1 im having the following error:
> 
> Details:
> 
> ```
> C:\Users\myproject\node_modules\microsoft-cognitiveservices-speech-sdk\distrib\lib\src\common.browser\PCMRecorder.js:85
>       this.privSpeechProcessorScript = new URL( /* webpackChunkName: 'script_processor_audioWorklet' */"speech-processor.js", import.meta.url).toString();
>                                                                                                                                      ^^^^
> 
> SyntaxError: Cannot use 'import.meta' outside a module
> 
> > 1 | import * as azureSdk from 'microsoft-cognitiveservices-speech-sdk';
>     | ^
> ```
> 
> I wanted to share this with you so you are aware! @glharper

For jest, you'll need a mock of PCMRecorder, as used in [this line](https://github.com/microsoft/cognitive-services-speech-sdk-js/blob/8af94521f9f55091afddd6431f39ba33a1189740/tests/AudioOutputStreamTests.ts#L10)

Our mock PCMRecorder is in [this directory](https://github.com/microsoft/cognitive-services-speech-sdk-js/blob/8af94521f9f55091afddd6431f39ba33a1189740/src/common.browser/__mocks__)

**Comment by fnematov** [comment]
Hello, I have used this lib with last version. I had same issue:

Error [ERR_REQUIRE_ESM]: require() of ES Module F:\Projects\myproject\node_modules\microsoft-cognitiveservices-speech-sdk\distrib\lib\microsoft.cognitiveservices.speech.sdk.js from F:\Projects\myproject\dist\azure-stt.js not supported.
Instead change the require of microsoft.cognitiveservices.speech.sdk.js in F:\Projects\myproject\dist\azure-stt.js to a dynamic import() which is available in all CommonJS modules.

I have installed "microsoft-cognitiveservices-speech-sdk": "^1.32.0", version and everything works fine

**Comment by zainzafar90** [comment]
> Maybe you installed `^1.32`, which effectively installs `1.33`.

TLDR; this worked for me

Although it should work for 1.33 but super-weird that it doesn't, I have created 3 projects, and every single time I have to remove this `^` from the package to make it work.

Can anybody explain why isn't 1.33 compatible anymore?

**Comment by glharper** [comment]
> According to #762 CJS support is planned to return.

Yes, the change back to CJS has been merged to master, and, unless release testing issues appear, will be in v1.34.0, due later this month.
